### PR TITLE
Remove mongoDB from ROSPlan demos (now param server) + code refactor

### PR DESCRIPTION
**Pull request by oscar-lima**
- remove mongoDB from ROSPlan demos
- code refactor and cleanup

### CODE REVIEW

**Comment by oscar-lima** [comment]
[ROSPlan mongoDB PR](https://github.com/KCL-Planning/ROSPlan/pull/197) needs to be merged at the same time.

**Review by gerardcanal — APPROVED** [review]

File: rosplan_turtlebot2_demo/config/waypoints.yaml [hunk 1]
```diff
@@ -0,0 +1,14 @@
+# waypoints reference frame
+waypoint_frameid: map
+
+# waypoint 2D coordinates: x, y, theta
+wp0: [0, 1, 2.5]
```
Comment (gerardcanal) [inline rosplan_turtlebot2_demo/config/waypoints.yaml]: Wouldn't it be better to have a super key "waypoints" with the wpX inside? That way we can easily iterate over them

File: rosplan_demos_interfaces/rosplan_interface_movebase/src/RPMoveBase.cpp [hunk 2]
```diff
@@ -1,102 +1,126 @@
...
+        // setup a move base clear costmap client (to be able to send clear costmap requests later on)
+        clear_costmaps_client_ = nh.serviceClient<std_srvs::Empty>("/move_base/clear_costmaps");
+    }
+
+    bool RPMoveBase::wpIDtoPoseStamped(std::string wpID, geometry_msgs::PoseStamped &result) {
+
+        ros::NodeHandle nh;
+        std::vector<double> wp;
+        if(nh.hasParam("/rosplan_demo_waypoints/" + wpID)) {
+            if(nh.getParam("/rosplan_demo_waypoints/" + wpID, wp)) {
```
Comment (gerardcanal) [inline rosplan_demos_interfaces/rosplan_interface_movebase/src/RPMoveBase.cpp:23]: (See coment in yaml) I see you actually have a super key, here, why isn't it the same in the yaml then?

File: rosplan_demos_interfaces/rosplan_interface_mapping/include/rosplan_interface_mapping/RPRoadmapServer.h [hunk 3]
```diff
@@ -1,170 +1,222 @@
...
-		bool generateRoadmap(rosplan_knowledge_msgs::CreatePRM::Request &req, rosplan_knowledge_msgs::CreatePRM::Response &res);
-		bool addWaypoint(rosplan_knowledge_msgs::AddWaypoint::Request &req, rosplan_knowledge_msgs::AddWaypoint::Response &res);
-		bool removeWaypoint(rosplan_knowledge_msgs::RemoveWaypoint::Request &req, rosplan_knowledge_msgs::RemoveWaypoint::Response &res);
-
-		void createPRM(nav_msgs::OccupancyGrid map, unsigned int nr_waypoints, double min_distance, double casting_distance, double connecting_distance, int occupancy_threshold, int total_attempts);
-	};
+    struct Waypoint
+    {
+        Waypoint(const std::string &id, unsigned int xCoord, unsigned int yCoord, const nav_msgs::MapMetaData& map_meta_data)
+            : wpID(id), grid_x(xCoord), grid_y(yCoord) {
```
Comment (gerardcanal) [inline rosplan_demos_interfaces/rosplan_interface_mapping/include/rosplan_interface_mapping/RPRoadmapServer.h]: I believe code should be moved to the cpps, it's not nice to have it between cpp and .h

File: rosplan_demos_interfaces/rosplan_interface_mapping/src/RPSimpleMapServer.cpp [hunk 4]
```diff
@@ -1,126 +1,191 @@
...
+        pose.pose.orientation.z = q.y();
+    }
+
+    bool RPSimpleMapServer::getWPCoordinates() {
+
+        // function that gets an unknown amount of waypoints from param server under a specific namespace
+
+        XmlRpc::XmlRpcValue waypoint_list;
+
+        if (!nh_.getParam("/waypoints", waypoint_list)) {
```
Comment (gerardcanal) [inline rosplan_demos_interfaces/rosplan_interface_mapping/src/RPSimpleMapServer.cpp]: Here the parameter is called something different

**Review by gerardcanal — CHANGES_REQUESTED** [review]

File: rosplan_demos_interfaces/rosplan_interface_mapping/src/RPSimpleMapServer.cpp (hunk 4 above)
Comment (oscar-lima) [reply rosplan_demos_interfaces/rosplan_interface_mapping/src/RPSimpleMapServer.cpp]: RPSimpleMapServer.cpp  removed as it is actually unused

File: rosplan_demos_interfaces/rosplan_interface_movebase/src/RPMoveBase.cpp (hunk 2 above)
Comment (oscar-lima) [reply rosplan_demos_interfaces/rosplan_interface_movebase/src/RPMoveBase.cpp:23]: launch file + yaml file slightly changed to enhance readability

File: rosplan_demos_interfaces/rosplan_interface_mapping/include/rosplan_interface_mapping/RPRoadmapServer.h (hunk 3 above)
Comment (oscar-lima) [reply rosplan_demos_interfaces/rosplan_interface_mapping/include/rosplan_interface_mapping/RPRoadmapServer.h]: feedback applied, thanks ;)

File: rosplan_turtlebot2_demo/config/waypoints.yaml (hunk 1 above)
Comment (oscar-lima) [reply rosplan_turtlebot2_demo/config/waypoints.yaml]: launch file + yaml file slightly changed to enhance readability

**Review by gerardcanal — APPROVED** [review]

**Pull request merged**